article/_posts/2021-07-13-CoopTrainingAndLatentSpaceDA.md

```diff
@@ -0,0 +1,89 @@
+the human brain (often referred to as systems 1 and 2 in the deep learning world). However, this simply boils down to a
+multi-task encoder-decoder network that performs both segmentation and reconstruction, followed by a segmentation
+autoencoder, with everything being trained end-to-end.
+
+![](/article/images/CoopTrainingAndLatentSpaceDA/figure1.jpg)
+
+## Latent space data augmentation
+The authors propose to randomly generate masks to apply to the latent space to drop features and generate corrupted, out
+of distribution data. In other words, they propose a form of dropout on the latent space vectors.
+
+Although not explained directly in the paper, the latent space learned seems to be structured such that
+$$z \in \mathbb{R}^{c \times h \times w}$$. The authors were thus able to use this structure to propose a few different
+schemes for generating the latent masks.
+
+1. **Random Masking with Dropout**: Drop random channels from the latent space.
+2. **Targeted Masking**: This scheme seeks to mask the most *salient* features, which can be determined by using two
+different methods. First, the gradients on the latent space have to be computed. Afterwards, either the i-th (i) pixels
+or (ii) channels with the highest mean gradients are masked. In this case, the features are not systemically dropped but
+rather they are scaled by an annealing factor $$a \in (0,0.5)$$.
+
+![](/article/images/CoopTrainingAndLatentSpaceDA/figure3.jpg)
+
+## Cooperative training
+By cooperative training, the authors mean that their method is trained on both standard data and data corrupted
+on-the-fly. Thus, the dual-network is exposed to three hard example pairs, i.e. corrupted images-clean images
+($$\hat{x}$$, $$x$$), corrupted images-GT ($$\hat{x}$$, $$y$$), and corrupted prediction-GT ($$\hat{p}$$, $$y$$). The
+overall loss of the network thus becomes $$\mathcal{L}_{cooperative} = \mathcal{L}_{std} + \mathcal{L}_{hard}$$, where
+$$\mathcal{L}_{hard}$$ is defined as:
+
+$$
+\mathcal{L}_{hard} = \mathbb{E}_{\hat{x},\hat{p},x,y} [ \mathcal{L}_{rec}(\mathcal{D}_{\phi_i}(E_{\theta}(\hat{x})),x) + \mathcal{L}_{seg}(\bar{p},y) + \mathcal{L}_{shp}(\mathcal{C}_{\psi}(\hat{p}),y) + \mathcal{L}_{shp}(\mathcal{C}_{\psi}(\bar{p}),y) ]
+$$
+
+where
+
+$$
+\bar{p} = \mathcal{D}_{\phi_i} (\mathcal{H}(E_{\theta}(\hat{x}))), \quad \text{the FTN's predicted segmentation on}~\hat{x}.
+$$
+
+
+# Data
+To test their method, the authors based themselves on the ACDC[^2] and M&Ms[^3] dataset. The method was trained on 10
+patient randomly sampled from ACDC's training set, and the multi-site M&Ms dataset only served for cross-domain test.
+The authors also used another 20 patients from the ACDC dataset, which they augmented with different types of MR
+artifacts to generate challenging intra-domain subsets.
+
+
+# Results
+
+![](/article/images/CoopTrainingAndLatentSpaceDA/table1.jpg)
```

Comment:
Looks this needs some explanation; compared methods seem on first column, datasets seem to be ACDC, M&M and corrupted versions (of ACDC according to the explanation above). However, the last two columns seem again to involve their method/their two networks. So it looks unclear to me how to read the table. Also, an explanation would be welcome for the difference in the method between the last two rows.

Reply:
I'll try to explain it better, but I must admit that I don't know how this table got past the reviewers :confused: Even having read the paper, the table is still confusing.

Reply:
OK, seems clearer after fc4f62a. Thanks.